### PR TITLE
Fix/double results in search 139

### PR DESCRIPTION
**Pull request by nicolasleger**
### What does/resolve this PR?

Trajet s'affichant deux fois dans une recherche

### How to deploy this PR by yourself to test it?

Testable sur https://cl-139.herokuapp.com/
Exemple https://cl-139.herokuapp.com/recherche?utf8=%E2%9C%93&search%5Bfrom_city%5D=Paris&search%5Bfrom_lon%5D=2.333333&search%5Bfrom_lat%5D=48.866667&search%5Bto_city%5D=Bordeaux&search%5Bto_lon%5D=-0.566667&search%5Bto_lat%5D=44.833333&search%5Bdate%5D=&commit=Partez+%21

`bundle exec rspec spec/models/trip_spec.rb`

### Issues resolved by this PR

https://github.com/covoiturage-libre/covoiturage-libre/issues/139

### CODE REVIEW

**Comment by maxb29** [comment]
Bonjour Nicolas,
Test rapide. le trajet qui s'affiche une seule fois c'est ok.
Par contre ce devrait être celui le plus proche de la ville recherchée.

Mon trajet:
image

Recherche _Bordeaux -> Paris_ (afficher l'annonce _bordeaux -> Paris_ plutot que _Artigues -> Paris_ )
image

File: app/models/trip.rb [hunk 1]
```diff
@@ -33,41 +34,78 @@ class Trip < ApplicationRecord
   default_scope { includes(:points).order('created_at ASC') }
 
   scope :from_to, -> (from_lon, from_lat, to_lon, to_lat) {
-    select('trips.*,
+    # Avoid Trips doublon
+    matching_points = Point.select("DISTINCT ON (point_a.trip_id) point_a.*,
```
Comment (webplate) [inline app/models/trip.rb]: I guess some factoring could be made between fromto fromonly and toonly. But it's really not mandatory.

File: spec/models/trip_spec.rb [hunk 2]
```diff
@@ -0,0 +1,93 @@
+require 'rails_helper'
+
+describe Trip, type: :model do
+
+  before(:all) do
+    city = City.create!
+
+    @trip = Trip.create!(
+      departure_date: Date.today,
```
Comment (webplate) [inline spec/models/trip_spec.rb]: Codacy suggest Time.zone.today

Comment (nicolasleger) [reply spec/models/trip_spec.rb]: Ok it doesn't matter in test I update it

File: app/models/trip.rb (hunk 1 above)
Comment (nicolasleger) [reply app/models/trip.rb]: Yes but I am bit worry that by making it generic it will be less understandable.

Comment (webplate) [reply app/models/trip.rb]: Yep, your right ! Let's keep it three cases.

**Comment by webplate** [comment]
@nicolasleger Do you have any idea how to implement the above proposal from @maxb29 ?

Certainly a good point !
As often ! :)

**Comment by nicolasleger** [comment]
Thanks @maxb29 and @webplate.
I updated the search to return the part with the least distance from both points.

Tests :

- https://cl-139.herokuapp.com/recherche?utf8=%E2%9C%93&search%5Bfrom_city%5D=&search%5Bfrom_lon%5D=&search%5Bfrom_lat%5D=44.833333&search%5Bto_city%5D=Lignan+de+Bordeaux&search%5Bto_lon%5D=-0.433333&search%5Bto_lat%5D=44.783333&search%5Bdate%5D=24%2F10%2F2017&commit=Partez+%21
- https://cl-139.herokuapp.com/recherche?utf8=%E2%9C%93&search%5Bfrom_city%5D=&search%5Bfrom_lon%5D=&search%5Bfrom_lat%5D=44.833333&search%5Bto_city%5D=Lignan+de+Bordeaux&search%5Bto_lon%5D=-0.433333&search%5Bto_lat%5D=44.783333&search%5Bdate%5D=24%2F10%2F2017&commit=Partez+%21
- https://cl-139.herokuapp.com/recherche?utf8=%E2%9C%93&search%5Bfrom_city%5D=Bordeaux&search%5Bfrom_lon%5D=-0.566667&search%5Bfrom_lat%5D=44.833333&search%5Bto_city%5D=&search%5Bto_lon%5D=&search%5Bto_lat%5D=48.866667&search%5Bdate%5D=24%2F10%2F2017&commit=Partez+%21
- https://cl-139.herokuapp.com/recherche?utf8=%E2%9C%93&search%5Bfrom_city%5D=Lignan+de+Bordeaux&search%5Bfrom_lon%5D=-0.433333&search%5Bfrom_lat%5D=44.783333&search%5Bto_city%5D=&search%5Bto_lon%5D=&search%5Bto_lat%5D=48.866667&search%5Bdate%5D=24%2F10%2F2017&commit=Partez+%21
- https://cl-139.herokuapp.com/recherche?utf8=%E2%9C%93&search%5Bfrom_city%5D=Lignan+de+Bordeaux&search%5Bfrom_lon%5D=-0.433333&search%5Bfrom_lat%5D=44.783333&search%5Bto_city%5D=Paris&search%5Bto_lon%5D=2.333333&search%5Bto_lat%5D=48.866667&search%5Bdate%5D=24%2F10%2F2017&commit=Partez+%21
- https://cl-139.herokuapp.com/recherche?utf8=%E2%9C%93&search%5Bfrom_city%5D=Bordeaux&search%5Bfrom_lon%5D=-0.566667&search%5Bfrom_lat%5D=44.833333&search%5Bto_city%5D=Paris&search%5Bto_lon%5D=2.333333&search%5Bto_lat%5D=48.866667&search%5Bdate%5D=24%2F10%2F2017&commit=Partez+%21

**Comment by webplate** [comment]
Such a pretty sight !!

**Pull request merged**